tests/unit/test_cmr_search.py

```diff
@@ -1,16 +1,28 @@
+from shutil import rmtree
+from tempfile import mkdtemp
 from unittest import TestCase
-from unittest.mock import patch
+from unittest.mock import patch, Mock
 
-from cmr import (GranuleQuery, CMR_UAT, CMR_OPS)
+from cmr import (GranuleQuery, CMR_UAT)
+import requests
+from requests.exceptions import HTTPError
 
-from varinfo.cmr_search import (get_granules, get_granule_link)
+from varinfo.cmr_search import (get_granules, get_granule_link,
+                                download_granule)
 from varinfo.exceptions import (CMRQueryException, MissingGranuleDownloadLinks,
-                                MissingPositionalArguments)
+                                MissingPositionalArguments, GranuleDownloadException)
 
 
 class TestQuery(TestCase):
     ''' A class for testing functions in cmr_search.
     '''
+    @classmethod
+    def setUp(self):
+        self.output_dir = mkdtemp()
+
+    def tearDown(self):
+        rmtree(self.output_dir)
```

Comment:
I think this is fine, but I am often a little wary of trying to use `rmtree` without first checking the directory exists. If there's anything in the source code that might delete the working directory, you run the risk of `rmtree` failing because it can't find what you asked to be deleted. I don't _think_ that's an issue with these tests, but a way to guard against it would be to import `os.path.exists` and then have:

```
def tearDown(self):
    if exists(self.output_dir):
        rmtree(self.output_dir)
```